includes/cache/wikia/CeleryPurge.class.php

```diff
@@ -71,6 +71,30 @@ static function onRestInPeace() {
 	}
 
 	/**
+	 * SUS-81: allow CDN purging by surrogate key
+	 *
+	 * Use OutputPage::tagWithSurrogateKeys() helper to emit proper headers
+	 *
+	 * @param string $key surrogate key to purge
+	 * @param string $service Fastly's service name (defaults to mediawiki)
+	 */
+	static function purgeBySurrogateKey( $key, $service = 'mediawiki' ) {
+		$caller = self::getPurgeCaller();
+		wfDebug( "Purging backtrace: " . wfGetAllCallers( false ) . "\n" );
+
+		WikiaLogger::instance()->info( 'varnish.purge', [
+			'key' => $key,
+			'service' => $service
+		] );
+
+		( new AsyncCeleryTask() )
+			->taskType( 'celery_workers.purger.purge' )
```

Comment:
Good call on introducing `EdgeCachePurger` task.

As for the difference between `purge` and `purgeBySurrogateKey` - the first one accepts URLs only and simply puts then in appropriate "buckets" for purge a proper Fastly service.